django_cloud_tasks/signals.py

```diff
@@ -0,0 +1,41 @@
+from django.core.exceptions import ValidationError
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+
+from django_cloud_tasks import models
+
+
+@receiver(pre_save, sender=models.Routine)
+def ensure_valid_task_name(sender, instance, **kwargs):
+    try:
+        instance.get_task()
+    except ValueError:
+        raise ValidationError(f"The task {instance.task_name} was not found. Make sure {instance.tasks_name} is properly set.")
+
+@receiver(pre_save, sender=models.Routine)
+def ensure_status_machine(sender, instance, **kwargs):
+    if not instance.pk:
+        return
```

Comment:
New routines should be checked too: I should not be able to create with `failed`, for instance.